nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java

```diff
@@ -0,0 +1,255 @@
+        this.url = null;
+    }
+
+    private void setupClient(ConfigurationContext context) throws Exception {
+        final String hosts = context.getProperty(HTTP_HOSTS).evaluateAttributeExpressions().getValue();
+        String[] hostsSplit = hosts.split(",[\\s]*");
+        this.url = hostsSplit[0];
+        final SSLContextService sslService =
+                context.getProperty(PROP_SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+        final String username = context.getProperty(USERNAME).evaluateAttributeExpressions().getValue();
+        final String password = context.getProperty(PASSWORD).evaluateAttributeExpressions().getValue();
+
+        final Integer connectTimeout = context.getProperty(CONNECT_TIMEOUT).asInteger();
+        final Integer readTimeout    = context.getProperty(SOCKET_TIMEOUT).asInteger();
+        final Integer retryTimeout   = context.getProperty(RETRY_TIMEOUT).asInteger();
+
+        HttpHost[] hh = new HttpHost[hostsSplit.length];
+        for (int x = 0; x < hh.length; x++) {
+            URL u = new URL(hostsSplit[x]);
+            hh[x] = new HttpHost(u.getHost(), u.getPort(), u.getProtocol());
+        }
+
+        RestClientBuilder builder = RestClient.builder(hh)
+                .setHttpClientConfigCallback(httpClientBuilder -> {
+                    if (sslService != null) {
+                        try {
+                            KeyStore keyStore = KeyStore.getInstance(sslService.getKeyStoreType());
+                            KeyStore trustStore = KeyStore.getInstance("JKS");
+
+                            try (final InputStream is = new FileInputStream(sslService.getKeyStoreFile())) {
+                                keyStore.load(is, sslService.getKeyStorePassword().toCharArray());
+                            }
+
+                            try (final InputStream is = new FileInputStream(sslService.getTrustStoreFile())) {
+                                trustStore.load(is, sslService.getTrustStorePassword().toCharArray());
+                            }
+
+                            final KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory
+                                    .getDefaultAlgorithm());
+                            kmf.init(keyStore, sslService.getKeyStorePassword().toCharArray());
+                            final TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory
+                                    .getDefaultAlgorithm());
+                            tmf.init(keyStore);
+                            SSLContext context1 = SSLContext.getInstance(sslService.getSslAlgorithm());
+                            context1.init(kmf.getKeyManagers(), tmf.getTrustManagers(), new SecureRandom());
+                            httpClientBuilder = httpClientBuilder.setSSLContext(context1);
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
```

Comment:
Why throw RTE here? (and below)

Reply:
Removed. I think I have a better solution for that now.